File: GuardianMonsters/core/src/de/limbusdev/guardianmonsters/fwmengine/battle/ui/widgets/BattleHUDMenuButton.java
package de.limbusdev.guardianmonsters.fwmengine.battle.ui.widgets;

import com.badlogic.gdx.scenes.scene2d.ui.ImageButton;
import com.badlogic.gdx.scenes.scene2d.ui.Skin;
import com.badlogic.gdx.utils.Align;

import de.limbusdev.guardianmonsters.enums.Element;

/**
 * BattleHUDMenuButton
 *
 * @author Georg Eckert 2017
 */

public class BattleHUDMenuButton extends ImageButton {

    public static final int X = 0, Y = 1;
    public static final int DEFEND=0, TEAM=1, BACK=2, BAG=3, SWORD=4, ESCAPE =5, EMPTY=6;
    public static final int ALIGN=Align.bottomLeft;

    public static final int[][] SLOTS = {
        {4,1}, {4,33}, {319,1}, {319,33}, {245,1}, {101,1}, {173,17}
    };

    public BattleHUDMenuButton(Skin skin, int position) {
        super(skin, construct(position));

        setPosition(SLOTS[position][X], SLOTS[position][Y], ALIGN);
    }

    private static String construct(int position) {
        String style;

        switch(position) {
            case DEFEND:
                style = "b-attack-extra";
                break;
            case TEAM:
                style = "b-attack-monsters";
                break;
            case BACK:
                style = "b-attack-back";
                break;
            case BAG:
                style = "b-attack-bag";
                break;
            case SWORD:
                style = "battle-fight";
                break;
            case ESCAPE:
                style = "battle-flee";
                break;
            default:    // EMPTY
                style = "battle-empty";
                break;
        }

        return style;
    }
}

File: GuardianMonsters/battle/src/de/limbusdev/guardianmonsters/battle/ui/widgets/BattleHUDMenuButton.kt
package de.limbusdev.guardianmonsters.battle.ui.widgets

import com.badlogic.gdx.scenes.scene2d.ui.ImageButton
import com.badlogic.gdx.scenes.scene2d.ui.Skin
import com.badlogic.gdx.utils.Align

/**
 * BattleHUDMenuButton
 *
 * @author Georg Eckert 2017
 */

class BattleHUDMenuButton(skin: Skin, position: Int) : ImageButton(skin, construct(position))
{
    // ................................................................................ Constructors
    init
    {
        setPosition(SLOTS[position][X].toFloat(), SLOTS[position][Y].toFloat(), ALIGN)
    }


    // ............................................................................ Companion Object
    companion object
    {
        const val X = 0
        const val Y = 1
        const val DEFEND = 0
        const val TEAM = 1
        const val BACK = 2
        const val BAG = 3
        const val SWORD = 4
        const val ESCAPE = 5
        const val EMPTY = 6
        const val ALIGN = Align.bottomLeft

        val SLOTS = arrayOf(

                intArrayOf(4, 1),
                intArrayOf(4, 33),
                intArrayOf(319, 1),
                intArrayOf(319, 33),
                intArrayOf(245, 1),
                intArrayOf(101, 1),
                intArrayOf(173, 17)
        )

        private fun construct(position: Int): String
        {
            return when (position)
            {
                DEFEND  -> "b-attack-extra"
                TEAM    -> "b-attack-monsters"
                BACK    -> "b-attack-back"
                BAG     -> "b-attack-bag"
                SWORD   -> "battle-fight"
                ESCAPE  -> "battle-flee"
                else    -> "battle-empty"
            }
        }
    }
}
